temporalio/ui
# Fix Boolean filter search

## Description & motivation 💭 
* Fixes filter search when the attribute type is `Bool`
* Adds `type` to `WorkflowFilter`
* Refactors filter type check utils to use `type` when possible (e.g. `isTextFilter`)

### Screenshots (if applicable) 📸 
![Screenshot 2023-12-08 at 11 20 36 AM](https://github.com/temporalio/ui/assets/15069288/f7d6ba67-56be-4a2f-95e6-4852a2555227)

### Design Considerations 🎨 

## Testing 🧪 

### How was this tested 👻 


### Steps for others to test: 🚶🏽‍♂️🚶🏽‍♀️ 
- Run a workflow with a custom search attribute of type `Bool`
- Filter by the custom search attribute
   - [ ] Verify there are no quotes around the filter value in the search params, filter input, or filter "chip" (e.g. `CustomBoolField=true` vs. `CustomBoolField="true"`)
- Refresh the page 
   - [ ] Verify there are no quotes around the filter value in the search params, filter input, or filter "chip" 
- Clip on the boolean filter and change the value 
   - [ ] Verify there are no quotes around the filter value in the search params, filter input, or filter "chip" 
## Checklists

### Draft Checklist 

### Merge Checklist 

### Issue(s) closed 
`DT-1659 `
## Docs

### Any docs updates needed?
